twittermap/public/javascripts/cache/services.js

```diff
@@ -0,0 +1,575 @@
+       //X from right to left
+
+        cutRegion(C_minX,C_minY,R_minX,C_maxY,user_direction).done(function(){
+
+                           deferred.resolve();
+        }).fail(function(){
+                ////console.log("X from left to right order minX,minY,maxX,maxY: ",R_minX,R_minY,R_maxX,R_maxY);
+                clearCache().done();
+                deferred.resolve();
+        })
+        return deferred.promise();
+
+   }else if(!LowerRight &&   !LowerLeft && !UpperLeft && !UpperRight){
+          //NO Overlap"
+          cutRegion(C_minX,C_minY,C_maxX,C_maxY,user_direction).done(function(){
+
+
+                 deferred.resolve();
+          }).fail(function(){
+                  ////console.log("NO OVERLAP .PROBLEM!");
+                  clearCache().done();
+                  deferred.resolve();
+          })
+          return deferred.promise();
+   }
+}
+/*Checks whether evicting some city polygons in a part of the cache region satisfies the target deletion count.
+If not, returns a failure message to the caller*/
+//Whether evicting the cached region vertically or horizontally. true - X axis (horizontally), false - Y axis (vertically)
+//TopTo Bottom = true
+//BottomToTop = false
+//LeftToRight = true
+//RightToLeft = false
+var cutRegion =  function findCornerofEviction(minX,minY,maxX,maxY,Direction){
+
+        var deferred = new $.Deferred();
+        var Xline,Yline;
+        if(Direction == "NE"){
+            Xline = turf.lineString([[minX,minY],[maxX,minY]]);
+            Yline = turf.lineString([[minX,minY],[minX,maxY]]);
+        }
+        if(Direction == "SW"){
+            Xline =  turf.lineString([[maxX,maxY],[minX,maxY]]);
+            Yline =  turf.lineString([[maxX,maxY],[maxX,minY]]);
+        }
+        if(Direction == "NW"){
+            Xline = turf.lineString([[maxX,minY],[minX,minY]]);
+            Yline = turf.lineString([[maxX,minY],[maxX,maxY]]);
+        }
+        if(Direction == "SE"){
+            Xline =  turf.lineString([[minX,maxY],[maxX,maxY]]);
+            Yline =  turf.lineString([[minX,maxY],[minX,minY]]);
+        }
+        var distanceX = turf.lineDistance(Xline, 'miles');
+        var distanceY = turf.lineDistance(Yline,'miles');
+        //Make 10 slices
+        var ShiftX = distanceX/10;
+        var ShiftY = distanceY/10;
+        var start = 0;
+        var stop= distanceX;
+        var removeItems;
+        var slicedXline,slicedYline,cutPointX,cutPointY,cutBbox,remove_search;
+        var moveX = ShiftX;
+        var moveY = ShiftY;
+        var ti = performance.now();
+        var count = 0;
+        while(deletedPolygonCount<targetDeleteCount){
+
+
+            slicedXline = turf.lineSliceAlong(Xline, start, moveX, 'miles');
+            slicedYline = turf.lineSliceAlong(Yline, start, moveY, 'miles');
+            cutPointX = slicedXline["geometry"]["coordinates"][1][0];
+            cutPointY = slicedYline["geometry"]["coordinates"][1][1];
+            if(Direction == "NE"){
```

Comment:
Trend = NE, Fathest Region = NE, we can’t delete SW, NE. Pick the farthest one from NW, SE.
Trend = SE, Fathest Region = SE, we can’t delete NW, SE. Pick the farthest one from NE, SW.
Trend = NW, Fathest Region = NW, we can’t delete SE...
Trend = SW, Fathest Region = SW, we can’t delete NE...